matrix/matrix.py

```diff
@@ -0,0 +1,193 @@
+                                      for i in (range(other.order))) for j in (range(other.order))))
+
+    def __mul__(self, other):
+        if type(other) is int or type(other) is float:
+            return Matrix(tuple(tuple(self.matrix[j][i] * other
+                                      for i in range(self.order)) for j in range(self.order)))
+
+        elif type(other) is Matrix:
+            if self.order != other.order:
+                raise MatrixNotLegalToMultiply(other)
+
+            return Matrix(tuple(tuple(sum(row[i] * cul[i] for i in range(self.order)) for cul in zip(*other))
+                                for row in self))
+        else:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+    def __rmul__(self, other):
+        if type(other) is not int and type(other) is not float:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+        return Matrix(tuple(tuple(self.matrix[j][i] * other
+                                  for i in range(self.order)) for j in range(self.order)))
+
+    def __sub__(self, other):
+        if type(other) is not Matrix:
+            raise ObjectTypeNotMatrix(other)
+
+        return self + (-other)
+
+    def __truediv__(self, other):
+        if type(other) is not int and type(other) is not float:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+        return Matrix(tuple(tuple(self.matrix[j][i] / other
+                                  for i in range(self.order)) for j in range(self.order)))
+
+    def __eq__(self, other):
+        if type(other) is not Matrix:
+            raise ObjectTypeNotMatrix(other)
+
+        if self.order != other.order:
+            return False
+
+        for i in range(self.order):
+            if self.matrix[i] != other.matrix[i]:
+                return False
+
+        return True
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __iter__(self):
+        object.__setattr__(self, "row_pointer", 0)
+        return self
+
+    def __next__(self):
+        if self.row_pointer >= self.order:
+            raise StopIteration
+        else:
+            result = self.matrix[self.row_pointer]
+            object.__setattr__(self, "row_pointer", self.row_pointer + 1)
+            return result
+
+    def __hash__(self):
+        return hash((tuple(self.matrix[j][i] for i in range(self.order) for j in range(self.order)), self.order))
+
+    @property
+    def tuples(self):
+        print("%s" % (self.matrix,))
+
+    def is_valid_matrix(self):
+        for row in self.matrix:
+            if type(row) is not tuple:
+                raise RowTypeNotTuple(row)
+            for item in row:
+                if type(item) is not int and type(item) is not float:
+                    raise RowItemNotNumber(item)
+            if len(row) != len(self.matrix):
+                raise RowAndColumnNumberDifferent(row, len(self.matrix))
```

Comment:
I think its way cleaner like:
```python
if any(type(row) is not tuple for row in self.matrix):
    raise TypeError("All rows should be tuples")

if any(any(type(item) is not int for item in row) for row in self.matrix):
    raise TypeError("Cells should be numbers")

if any(len(self.matrix) != len(row) for row in self.matrix):
    raise ValueError("Number of rows and columns should be the same")
```

Also, There are alot of builtin exception in python that pretty much cover all the basic uses of exceptions there is not need to create more.